specification/indices/modify_data_stream/types.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { DataStreamName, IndexName } from '@_types/common'
+
+/** @variants container */
+export class Action {
+  add_backing_index?: AddBackingIndexAction
+  remove_backing_index?: RemoveBackingIndexAction
+}
+
+export class AddBackingIndexAction {
+  index: IndexName
+  data_stream: DataStreamName
+}
+
+export class RemoveBackingIndexAction {
+  index: IndexName
+  data_stream: DataStreamName
+}
```

Comment:
Isn't this more forward-thinking though since they're two different actions? In the future there may be additional properties added to one action but not the other.

Reply:
Yes it is, but until that problem presents I think we should keep the "type proliferation" as small as possible.
We are reusing the same type in a similar manner in other apis as well.